beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -1258,16 +1256,47 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         }
     }
 
-    // Update peers per client metrics.
-    fn update_peers_per_client_metrics(&self) {
+    // Update peer count related metrics.
+    fn update_peer_count_metrics(&self) {
+        let mut peers_connected = 0;
         let mut clients_per_peer = HashMap::new();
+        let mut peers_connected_mutli: HashMap<(&str, &str), i32> = HashMap::new();
+
+        for (_, peer_info) in self.network_globals.peers.read().connected_peers() {
+            peers_connected += 1;
 
-        for (_peer, peer_info) in self.network_globals.peers.read().connected_peers() {
             *clients_per_peer
                 .entry(peer_info.client().kind.to_string())
                 .or_default() += 1;
+
+            let direction = match peer_info.connection_direction() {
+                Some(ConnectionDirection::Incoming) => "inbound",
+                Some(ConnectionDirection::Outgoing) => "outbound",
+                None => "none",
+            };
+            // Note: the `transport` is set to `unknown` if the `listening_addresses` list is empty.
+            // This situation occurs when the peer is initially registered in PeerDB, but the peer
```

Comment:
thanks for researching this Akihito! But since we get the Multiaddress as soon as the peer connects, can't we add it immediately? Don't worry doesn't need to be address in this PR, I think I plan to submit a subsequent PR cause  we also  know the direction as soon as the peer connects, so it doesn't make sense `connection_direction()` be `None`

Reply:
> But since we get the Multiaddress as soon as the peer connects, can't we add it immediately?

Yes, I believe we can do that, but I'm not very familiar with PeerDB. However I've delved a little into the PeerDB implementation recently. During this exploration, I discovered that the Multiaddress obtained when a peer connects is stored in `PeerInfo::seen_multiaddrs`. The `seen_multiaddrs` is stored into PeerInfo [here](https://github.com/sigp/lighthouse/blob/f919f82e4f7eb06baa2fa3a41e8ec8951fccdabc/beacon_node/lighthouse_network/src/peer_manager/peerdb.rs#L757-L761).

https://github.com/sigp/lighthouse/blob/d36241b4a12719800d04673d8f700ccf0d9253a8/beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs#L32-L34

Given this understanding, would it be appropriate to iterate over `seen_multiaddrs` in cases where `listening_addresses()` returns empty?  This approach could potentially ensure that we always have an address, even immediately after a peer connects.
cc @AgeManning